### PR TITLE
feat(ELB): add elb loadbalancers data source

### PR DESCRIPTION
**Pull request by SabreKeyZ**
**What this PR does / why we need it**:
add elb loadbalancers data source
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
add elb loadbalancers data source
```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST=./huaweicloud/services/acceptance/elb/ TESTARGS='-run TestAccDatasourceLoadBalancers_basic' 
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/elb/ -v -run TestAccDatasourceLoadBalancers_basic -timeout 360m -parallel 4 
=== RUN   TestAccDatasourceLoadBalancers_basic 
=== PAUSE TestAccDatasourceLoadBalancers_basic
=== CONT  TestAccDatasourceLoadBalancers_basic
--- PASS: TestAccDatasourceLoadBalancers_basic (57.73s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/elb       57.773s

```

### CODE REVIEW

File: docs/data-sources/elb_loadbalancers.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+## Attribute Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `loadbalancers` - Lists the load balancers.
+  The [object](#loadbalancers_object) structure is documented below.
+
+<a name="loadbalancers_object"></a>
+The `loadbalancers` block supports:
```
Comment (houpeng80) [inline docs/data-sources/elb_loadbalancers.md]:
```suggestion
* `loadbalancers` - Lists the load balancers.
  The [loadbalancers](#Elb_loadbalancer_loadbalancers) structure is documented below.

<a name="Elb_loadbalancer_loadbalancers"></a>
The `loadbalancers` block supports:
```

File: docs/data-sources/elb_loadbalancers.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+* `autoscaling_enabled` - Whether the current load balancer enables elastic expansion.
+
+* `backend_subnets` - Lists the IDs of subnets on the downstream plane.
+
+* `protection_status` - Modify the protection status, value: nonProtection: No protection, the default value is  
+  nonProtection, consoleProtection: console modification protection.
+
+* `protection_reason` - Reasons for setting protection, illustrate: Only valid when protection_status is
+  consoleProtection, Minimum length=0, Maximum length=255.
```
Comment (houpeng80) [inline docs/data-sources/elb_loadbalancers.md]: modify it according to the document of resources/elb_loadbalancer.md

File: huaweicloud/services/elb/data_source_huaweicloud_elb_loadbalancers.go [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	"github.com/chnsz/golangsdk/pagination"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+func DataSourceElbLoadbalancesV3() *schema.Resource {
```
Comment (ShiChangkuo) [inline huaweicloud/services/elb/data_source_huaweicloud_elb_loadbalancers.go]: the function name should not end with **v3**, please check other places.

File: huaweicloud/services/elb/data_source_huaweicloud_elb_loadbalancers.go [hunk 4]
```diff
@@ -0,0 +1,297 @@
...
+	if v, ok := d.GetOk("loadbalancer_id"); ok {
+		res = fmt.Sprintf("%s&id=%v", res, v)
+	}
+	if v, ok := d.GetOk("name"); ok {
+		res = fmt.Sprintf("%s&name=%v", res, v)
+	}
+	if v, ok := d.GetOk("description"); ok {
+		res = fmt.Sprintf("%s&description=%v", res, v)
+	}
+	if v, ok := d.GetOk("share_type"); ok {
```
Comment (ShiChangkuo) [inline huaweicloud/services/elb/data_source_huaweicloud_elb_loadbalancers.go]: it's better to add an acceptance test for `share_type` param.

File: huaweicloud/services/elb/data_source_huaweicloud_elb_loadbalancers.go [hunk 5]
```diff
@@ -0,0 +1,297 @@
...
+			},
+			"l4_flavor_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"l7_flavor_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"region": {
```
Comment (ShiChangkuo) [inline huaweicloud/services/elb/data_source_huaweicloud_elb_loadbalancers.go]: region is not necessary in the object.

File: huaweicloud/services/elb/data_source_huaweicloud_elb_loadbalancers.go [hunk 6]
```diff
@@ -0,0 +1,305 @@
...
+			"autoscaling_enabled": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"min_l7_flavor_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"backend_subnets": {
+				Type:     schema.TypeSet,
```
Comment (ShiChangkuo) [inline huaweicloud/services/elb/data_source_huaweicloud_elb_loadbalancers.go]: it's better to use **TypeList**

File: huaweicloud/services/elb/data_source_huaweicloud_elb_loadbalancers.go [hunk 7]
```diff
@@ -0,0 +1,305 @@
...
+			"type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+	return &sc
+}
+
+func dataSourceElbLoadBalancersV3Read(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```
Comment (ShiChangkuo) [inline huaweicloud/services/elb/data_source_huaweicloud_elb_loadbalancers.go]:
```suggestion
func dataSourceElbLoadBalancersRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```

**Pull request merged**